ᴡɪɴᴅᴏᴡs:  Link GLFW appropriately with CMake

It seems like this issue is not fully completed as CMake is failing to link important libraries:

```shell
...
[100%] Linking C executable /home/runner/work/oc-cplatform/oc-cplatform/bin/oc_win32
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0x4b): undefined reference to `glfwInit'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0x91): undefined reference to `glfwCreateWindow'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xa4): undefined reference to `glfwTerminate'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xb8): undefined reference to `glfwMakeContextCurrent'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xc7): undefined reference to `glfwSwapBuffers'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xcc): undefined reference to `glfwPollEvents'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xd9): undefined reference to `glfwWindowShouldClose'
/usr/bin/i686-w64-mingw32-ld: CMakeFiles/oc_win32.dir/src/main.c.o:main.c:(.text+0xe2): undefined reference to `glfwTerminate'
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/oc_win[32](https://github.com/CiroZDP/oc-cplatform/actions/runs/13590517124/job/37995242404#step:5:33).dir/build.make:101: /home/runner/work/oc-cplatform/oc-cplatform/bin/oc_win32] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:122: CMakeFiles/oc_win32.dir/all] Error 2
...
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.10)
 project(OpenCraft C)
 
+set(CMAKE_VERBOSE_MAKEFILE ON)
+
 # Set C standard
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED True)

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -57,5 +57,4 @@ target_link_libraries(opencraft PRIVATE
         -static
         -static-libgcc
         -static-libstdc++
-        -mwindows
 )

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -46,15 +46,8 @@ endif()
 # Build executable
 add_executable(opencraft ${SRC_FILES})
 
-# Link GLFW statically
-target_link_libraries(opencraft PRIVATE
-        ${GLFW_STATIC_LIB}
-        -L${GLFW_ARCH_DIR}
-        -lglfw3
-        -lopengl32
-        -lgdi32
-        -lwinmm
-        -static
-        -static-libgcc
-        -static-libstdc++
-)
+link_directories(${GLFW_ARCH_DIR})
+target_compile_options(opencraft PRIVATE -static)
+
+# Link libraries
+target_link_libraries(opencraft PRIVATE ${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -38,6 +38,7 @@ set(CMAKE_EXECUTABLE_SUFFIX ".exe")
 
 set(GLFW_STATIC_LIB "${GLFW_ARCH_DIR}/libglfw3.a")
 
+# Test if GLFW exists
 if(NOT EXISTS ${GLFW_STATIC_LIB})
     message(FATAL_ERROR "GLFW static library not found: ${GLFW_STATIC_LIB}")
 endif()
@@ -46,9 +47,8 @@ endif()
 add_executable(opencraft ${SRC_FILES})
 
 # Link GLFW statically
-target_compile_options(opencraft PRIVATE -static)
-
-target_link_options(opencraft PRIVATE
+target_link_libraries(opencraft PRIVATE
+        ${GLFW_STATIC_LIB}
         -L${GLFW_ARCH_DIR}
         -lglfw3
         -lopengl32
@@ -59,4 +59,3 @@ target_link_options(opencraft PRIVATE
         -static-libstdc++
         -mwindows
 )
-

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -54,5 +54,5 @@ add_executable(opencraft ${SRC_FILES})
 target_compile_options(opencraft PRIVATE -static)
 
 # Link libraries
-link_libraries(opencraft PRIVATE ${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)
+link_libraries(${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)
 target_link_libraries(opencraft PRIVATE ${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -35,6 +35,20 @@ set(CMAKE_EXECUTABLE_SUFFIX ".exe")
 # Build executable
 add_executable(opencraft ${SRC_FILES})
 
+### BEGIN TESTS ###
+set(GLFW_STATIC_LIB "${GLFW_ARCH_DIR}/libglfw3.a")
+set(GLFW_DLL_LIB "${GLFW_ARCH_DIR}/libglfw3dll.a")
+
+if(NOT EXISTS ${GLFW_STATIC_LIB})
+    message(FATAL_ERROR "GLFW static library not found: ${GLFW_STATIC_LIB}")
+endif()
+
+if(NOT EXISTS ${GLFW_DLL_LIB})
+    message(FATAL_ERROR "GLFW DLL library not found: ${GLFW_DLL_LIB}")
+endif()
+
+### END TESTS ###
+
 # Link GLFW statically
 target_link_libraries(opencraft PRIVATE
         ${GLFW_ARCH_DIR}/libglfw3.a

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -48,7 +48,6 @@ add_executable(opencraft ${SRC_FILES})
 
 # Ensure to locate the directory where GLFW is
 link_directories(${GLFW_ARCH_DIR})
-target_link_directories(${GLFW_ARCH_DIR})
 
 # Static linking
 ## user won't require additional dlls' ;)

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -46,15 +46,17 @@ endif()
 add_executable(opencraft ${SRC_FILES})
 
 # Link GLFW statically
-target_link_libraries(opencraft PRIVATE
-        ${GLFW_STATIC_LIB}
+target_compile_options(opencraft PRIVATE -static)
+
+target_link_options(opencraft PRIVATE
+        -L${GLFW_ARCH_DIR}
+        -lglfw3
+        -lopengl32
+        -lgdi32
+        -lwinmm
         -static
         -static-libgcc
         -static-libstdc++
         -mwindows
-        opengl32
-        gdi32
-        user32
-        shell32
 )
 

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -19,6 +19,7 @@ if(TARGET_WINDOWS_64)
 
     # GLFW
     set(GLFW_ARCH_DIR "${GLFW_DIR}/win64")
+
 elseif(TARGET_WINDOWS_32)
     message(STATUS "Building for Windows 32-bit")
     set(CMAKE_SYSTEM_NAME Windows)
@@ -28,36 +29,32 @@ elseif(TARGET_WINDOWS_32)
 
     # GLFW
     set(GLFW_ARCH_DIR "${GLFW_DIR}/win32")
+
+else()
+    message(FATAL_ERROR "No target architecture specified. Use -DTARGET_WINDOWS_64=ON or -DTARGET_WINDOWS_32=ON")
 endif()
 
 set(CMAKE_EXECUTABLE_SUFFIX ".exe")
 
-# Build executable
-add_executable(opencraft ${SRC_FILES})
-
-### BEGIN TESTS ###
 set(GLFW_STATIC_LIB "${GLFW_ARCH_DIR}/libglfw3.a")
-set(GLFW_DLL_LIB "${GLFW_ARCH_DIR}/libglfw3dll.a")
 
 if(NOT EXISTS ${GLFW_STATIC_LIB})
     message(FATAL_ERROR "GLFW static library not found: ${GLFW_STATIC_LIB}")
 endif()
 
-if(NOT EXISTS ${GLFW_DLL_LIB})
-    message(FATAL_ERROR "GLFW DLL library not found: ${GLFW_DLL_LIB}")
-endif()
-
-### END TESTS ###
+# Build executable
+add_executable(opencraft ${SRC_FILES})
 
 # Link GLFW statically
 target_link_libraries(opencraft PRIVATE
-        ${GLFW_ARCH_DIR}/libglfw3.a
+        ${GLFW_STATIC_LIB}
         -static
         -static-libgcc
         -static-libstdc++
         -mwindows
-        -lopengl32
-        -lgdi32
-        -luser32
-        -lshell32
+        opengl32
+        gdi32
+        user32
+        shell32
 )
+

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -46,8 +46,14 @@ endif()
 # Build executable
 add_executable(opencraft ${SRC_FILES})
 
+# Ensure to locate the directory where GLFW is
 link_directories(${GLFW_ARCH_DIR})
+target_link_directories(${GLFW_ARCH_DIR})
+
+# Static linking
+## user won't require additional dlls' ;)
 target_compile_options(opencraft PRIVATE -static)
 
 # Link libraries
+link_libraries(opencraft PRIVATE ${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)
 target_link_libraries(opencraft PRIVATE ${GLFW_STATIC_LIB} glfw3 opengl32 gdi32 winmm)

--- a/cmake/Windows.cmake
+++ b/cmake/Windows.cmake
@@ -43,11 +43,11 @@ if(NOT EXISTS ${GLFW_STATIC_LIB})
     message(FATAL_ERROR "GLFW static library not found: ${GLFW_STATIC_LIB}")
 endif()
 
-# Build executable
-add_executable(opencraft ${SRC_FILES})
-
 # Ensure to locate the directory where GLFW is
 link_directories(${GLFW_ARCH_DIR})
+
+# Build executable
+add_executable(opencraft ${SRC_FILES})
 
 # Static linking
 ## user won't require additional dlls' ;)

